pkg/streamingpromql/compat/fallback_header_test.go

```diff
@@ -43,7 +45,11 @@ func TestEngineFallbackInjector(t *testing.T) {
 
 	for name, testCase := range testCases {
 		t.Run(name, func(t *testing.T) {
-			injector := EngineFallbackInjector{}
+			injector := api.HeaderOptionsMiddleware(api.HeaderOptionsConfig{
+				Headers: map[string]func(string) error{
+					ForceFallbackHeaderName: ForceFallbackHeaderNameValidator,
+				},
+			})
```

Comment:
Could we use or somehow share this config here and the config that's configured in `DefaultHeaderOptionsMiddlewareConfig`? Then we don't have to worry about one getting out of sync with the other